fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -458,6 +468,309 @@ Value makePadding(int size) {
 	return pad.substr(0, size);
 }
 
+struct IRangeFileWriter {
+public:
+	virtual Future<Void> padEnd(bool final) = 0;
+
+	virtual Future<Void> writeKV(Key k, Value v) = 0;
+
+	virtual Future<Void> writeKey(Key k) = 0;
+
+	virtual Future<Void> finish() = 0;
+
+	virtual ~IRangeFileWriter() {}
+};
+
+struct SnapshotFileBackupCipherKey {
+	EncryptCipherDomainId encryptDomainId;
+	EncryptCipherBaseKeyId baseCipherId;
+	EncryptCipherRandomSalt salt;
+	StringRef baseCipher;
+
+	static SnapshotFileBackupCipherKey fromBlobCipherKey(Reference<BlobCipherKey> keyRef, Arena& arena) {
+		SnapshotFileBackupCipherKey cipherKey;
+		cipherKey.encryptDomainId = keyRef->getDomainId();
+		cipherKey.baseCipherId = keyRef->getBaseCipherId();
+		cipherKey.salt = keyRef->getSalt();
+		cipherKey.baseCipher = makeString(keyRef->getBaseCipherLen(), arena);
+		memcpy(mutateString(cipherKey.baseCipher), keyRef->rawBaseCipher(), keyRef->getBaseCipherLen());
+
+		return cipherKey;
+	}
+};
+
+struct SnapshotFileBackupCipherKeysCtx {
+	SnapshotFileBackupCipherKey textCipherKey;
```

Comment:
I think code here is mimicking BlobGranule file encryption model; however, usecases may differ. 
For BlobGranule encryption, the actual encryption/decryption is done in client context which might not support `ACTOR` or `wait` model; hence, I did the following:
1. BlobWorker fetches the encryption keys - interacting with EKP
2. Construct in-memory encryption key representation and then pass to BlobGranueFile and other client context to enable re-construction of key without needing to access EKP. 

Do we think we have similar scenario here?

Reply:
Ah okay i didn't realize this was the reason for it, in that case we can simplify the logic